### PR TITLE
crimson/os/seastore/backref_manager: scan backref entries by journal seq

### PR DESCRIPTION
**Pull request by zhscn**
Scanning backref entries by paddr might cause errors at SpaceTracker:

https://github.com/ceph/ceph/blob/2f2c9aa739d4df0f0e1d045193f8408d6acd4b37/src/crimson/os/seastore/async_cleaner.cc#L718-L743

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/os/seastore/backref/btree_backref_manager.cc [hunk 1]
```diff
@@ -397,34 +397,37 @@ BtreeBackrefManager::scan_mapped_space(
       );
     }).si_then([this, &scan_visitor, c, FNAME, block_size] {
       // traverse alloc-deltas in order
-      auto &backref_entry_mset = cache.get_backref_entry_mset();
```
[the remark this answers is not in the document]
Comment (xxhdx1985126) [reply src/crimson/os/seastore/backref/btree_backref_manager.cc:400]: @cyx1231st The following alloc/free sequence may lead to invalid space usage status:

1. alloc extent `A` at paddr `P1` with length `L1`
2. alloc extent `B` at paddr `P2=P1+L1` with length `L2`
3. free `A` and `B`
4. alloc extent `C` at paddr `P1` with length `L3 = L1 + L2`

If we follow the `backref_entry_mset` to replay the backrefs, the alloc/free sequence that are replayed may be as follows:

1. alloc extent `A` at paddr `P1` with length `L1`
2. free `A`
3. alloc extent `C` at paddr `P1` with length `L3`;
4. alloc extent `B` at paddr `P2` with length `L2`;
5. free extent `B`

Although the final result is correct, the space usage status at step 3 may be invalid, am I right? @zhscn

[the remark this answers is not in the document]
Comment (cyx1231st) [reply src/crimson/os/seastore/backref/btree_backref_manager.cc:400]: Make sense to me. Thanks for clarifying!

**Review by cyx1231st — APPROVED** [review]
LGTM!

Please associate this fix in https://tracker.ceph.com/projects/crimson/issues for backporting.

**Comment by cyx1231st** [comment]
https://tracker.ceph.com/issues/61480

**Pull request merged**